pennylane/templates/state_preparations/mottonen.py

```diff
@@ -87,21 +86,22 @@ def _compute_theta(alpha):
     k = np.log2(alpha.shape[0])
     factor = 2 ** (-k)
 
-    theta = sparse.dok_matrix(alpha.shape, dtype=np.float64)  # type: sparse.dok_matrix
+    theta = np.zeros(alpha.shape, dtype=np.float64)
 
     for row in range(alpha.shape[0]):
         # Use transpose of M:
-        entry = sum([_matrix_M_entry(col, row) * a for (col, _), a in alpha.items()])
```

Comment:
This is what happened here before: 

all matrices, like alpha, were sparse matrices representing vectors, which means that the second dimension was a dummy (hence `(col, _)`).